runtime/common/src/main/kotlin/kotlinx/serialization/json/internal/TreeJsonInput.kt

```diff
@@ -0,0 +1,183 @@
+@file:Suppress("LeakingThis")
+
+package kotlinx.serialization.json.internal
+
+import kotlinx.serialization.*
+import kotlinx.serialization.internal.*
+import kotlinx.serialization.json.*
+
+internal fun <T> Json.readJson(element: JsonElement, deserializer: DeserializationStrategy<T>): T {
+    val descriptor = deserializer.descriptor
+    if (element is JsonNull) {
+        // TODO temporary workaround (?)
+        require(descriptor.isNullable) { "Read JsonNull and expected nullable descriptor, but has $descriptor" }
+        @Suppress("NULL_FOR_NONNULL_TYPE")
```

Comment:
Are you sure that this suppress can't trigger e.g. SEGFAULT on Native or something undesirable on other platforms?

Reply:
Yes, I have tests for it.
It will be fixed around #287